### PR TITLE
[WFCORE-2502] - Legacy ldap realm,entry for non existing user are cached

### PR DESCRIPTION
**Pull request by JiriOndrusek**
WFCORE: https://issues.jboss.org/browse/WFCORE-2502
Downstream (7.2): https://issues.jboss.org/browse/JBEAP-9542
Downstream (7.1): https://issues.jboss.org/browse/JBEAP-9391
Downstream (7.0): https://issues.jboss.org/browse/JBEAP-11350

### CODE REVIEW

**Comment by bstansberry** [comment]
this is ok to test

File: domain-management/src/test/java/org/jboss/as/domain/management/security/LdapCacheServiceMockTest.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
+package org.jboss.as.domain.management.security;
```
Comment (JiriOndrusek) [inline domain-management/src/test/java/org/jboss/as/domain/management/security/LdapCacheServiceMockTest.java]: Header was added.

**Comment by JiriOndrusek** [comment]
retest this please

**Comment by bstansberry** [comment]
@JiriOndrusek Thanks. Can you squash this to a single commit?

**Comment by JiriOndrusek** [comment]
@bstansberry commits are squashed.

**Pull request merged**